pm/sender.go

```diff
@@ -74,6 +78,24 @@ func (s *sender) EV(sessionID string) (*big.Rat, error) {
 	return ticketEV(session.ticketParams.FaceValue, session.ticketParams.WinProb), nil
 }
 
+func (s *sender) Validate() error {
```

Comment:
Instead of adding an additional method to the `Sender` public API we could move these checks into an internal method which is called in `CreateTicketBatch()` (similar to how we do the ticket params checks):

```go
func (s *sender) CreateTicketBatch(...) (*TicketBatch, error) {
    ...

    if err := s.validateSender(); err != nil {
        return err
    }

    if err := s.validateTicketParams(&session.ticketParams, size); err != nil {
        return err
    }

    ...
}
```

I see the benefits of this approach as:

- Eliminates the need to modify the `Sender` public API
- Removes some code from `mediaserver.go`
- Runs sender validation mid-stream with each payment that B prepares for an O for a stream instead of just running sender validation at the beginning of a stream when registering the RTMP connection

The two downsides of this approach are:
- The RTMP connection will not be closed immediately if one of these sender validation checks would fail because the checks are not run until after the RTMP connection is already established
- At the moment, the RTMP connection will not be closed mid-stream if one of these sender validation checks fail because we [always retry segments and do not propagate/handle errors from `transcodeSegment()`](https://github.com/livepeer/go-livepeer/blob/master/server/broadcast.go#L279)

I think the first downside is a reasonable tradeoff - we incur a little extra cost for RTMP connection setup, but we get to isolate the sender validation logic in the `pm` package which will be executed whenever a ticket batch is being prepared via `CreateTicketBatch()`.

As for the second downside, outside of this PR, we could introduce a non-retryable error as suggested by this [comment](https://github.com/livepeer/go-livepeer/blob/master/server/broadcast.go#L298). If a non-retryable error, such as failing sender validation checks), is encountered then the error would be propagated from `transcodeSegment()` up to the caller of `processSegment()` which can then close the RTMP connection to alert the user. Prior to the introduction of a non-retryable error, B would just keep logging errors from sender validation as it continually retries segments (without actually submitting the segments).

Reply:
fixed in 35979be but will need to track the extra PR needed in a seperate issue.

Reply:
@kyriediculous Can you create an issue to track the additional work needed to close the RTMP connections when the `validateSender()` check fails?